### PR TITLE
Simple move of the OAuth clients bootstrap

### PR DESCRIPTION
**Pull request by stlaz**
Move oauthclients bootstrap from the integrated OAuth server

------
This is to unblock any work that depends on OAuth clients bootstrap being done separately from the integrated OAuth server running.

I had trouble creating a cluster today so it's not yet tested.

The code should eventually be replaced with the outcome of https://github.com/openshift/cluster-authentication-operator/pull/136

### CODE REVIEW

**Comment by sttts** [comment]
"I had trouble creating a cluster today so it's not yet tested."

Should it before merging?

**Comment by sttts** [comment]
> Move oauthclients bootstrap from the integrated OAuth server

Link?

File: Dockerfile [hunk 1]
```diff
@@ -1,9 +1,9 @@
-FROM registry.svc.ci.openshift.org/openshift/release:golang-1.10 AS builder
+FROM registry.svc.ci.openshift.org/openshift/release:golang-1.12 AS builder
```
Comment (enj) [inline Dockerfile]: Update `Dockerfile.rhel` also

Comment (stlaz) [reply Dockerfile]: Done

**Comment by stlaz** [comment]
@sttts It was a problem with the installer ex

> "I had trouble creating a cluster today so it's not yet tested."
> 
> Should it before merging?

It was trouble with the installer, it was timing out w/o this patch in, I believe I saw the issue reported on Friday already, after building the fresh master I was able to get a cluster eventually.

The link for the moved code is now to be found in the origin PR that I linked to this one

File: pkg/operator2/oauthclients.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+	"k8s.io/client-go/util/retry"
+
+	oauthv1 "github.com/openshift/api/oauth/v1"
+	oauthclient "github.com/openshift/client-go/oauth/clientset/versioned/typed/oauth/v1"
+	"github.com/openshift/library-go/pkg/oauth/oauthdiscovery"
+)
+
+// EnsureBootstrapOAuthClients creates or updates the bootstrap oauth clients that openshift relies upon.
+func (c *authOperator) ensureBootstrappedOAuthClients(masterPublicURL string) error {
+	const browserSecretByteLen = 32
```
Comment (enj) [inline pkg/operator2/oauthclients.go]: dead code

File: pkg/operator2/oauthclients.go [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+		existingCopy := existing.DeepCopy()
+
+		if len(existingCopy.Secret) == 0 {
+			existingCopy.Secret = client.Secret
+		}
+
+		existingCopy.RedirectURIs = client.RedirectURIs
+
+		// If the GrantMethod is present, keep it for compatibility
```
Comment (enj) [inline pkg/operator2/oauthclients.go]: we probably do not want this

File: pkg/operator2/oauthclients.go [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		existing, err := oauthClients.Get(client.Name, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+
+		existingCopy := existing.DeepCopy()
+
+		if len(existingCopy.Secret) == 0 {
```
Comment (enj) [inline pkg/operator2/oauthclients.go]: this actually need a bit more nuance but is good enough for now

**Comment by enj** [comment]
/lgtm

All this code should get deleted this week anyway so let us not block others.

**Comment by enj** [comment]
/lgtm cancel

Seems like this breaks kube:admin

**Comment by enj** [comment]
#139 has only minor changes on top of this PR.

/close

**Pull request closed without merging**